Assets/Scripts/Gameplay/GameplayObjects/Character/ServerCharacter.cs

```diff
@@ -92,34 +176,62 @@ public override void OnNetworkSpawn()
 
         public override void OnNetworkDespawn()
         {
-            if (NetState)
-            {
-                NetState.DoActionEventServer -= OnActionPlayRequest;
-                NetState.ReceivedClientInput -= OnClientMoveRequest;
-                NetState.OnStopChargingUpServer -= OnStoppedChargingUp;
-                NetState.NetworkLifeState.LifeState.OnValueChanged -= OnLifeStateChanged;
-            }
+            NetLifeState.LifeState.OnValueChanged -= OnLifeStateChanged;
 
             if (m_DamageReceiver)
             {
-                m_DamageReceiver.damageReceived -= ReceiveHP;
-                m_DamageReceiver.collisionEntered -= CollisionEntered;
+                m_DamageReceiver.DamageReceived -= ReceiveHP;
+                m_DamageReceiver.CollisionEntered -= CollisionEntered;
             }
         }
 
+
+        /// <summary>
+        /// RPC to send inputs for this character from a client to a server.
+        /// </summary>
+        /// <param name="movementTarget">The position which this character should move towards.</param>
+        [ServerRpc]
+        public void SendCharacterInputServerRpc(Vector3 movementTarget)
+        {
+            OnClientMoveRequest(movementTarget);
```

Comment:
now that these are in the same file, do we need the extra method? OnClientMoveRequest is only called once here

Reply:
same with other RPCs